core/src/main/java/com/radixdlt/consensus/bft/VertexStoreJavaImpl.java

```diff
@@ -360,10 +369,15 @@ private Optional<CommittedUpdate> commit(BFTHeader header, QuorumCertificate com
   public LinkedList<ExecutedVertex> getPathFromRoot(HashCode vertexId) {
     final LinkedList<ExecutedVertex> path = new LinkedList<>();
 
-    ExecutedVertex vertex = vertices.get(vertexId);
-    while (vertex != null) {
-      path.addFirst(vertex);
-      vertex = vertices.get(vertex.getParentId());
+    var executedVertexOpt = getExecutedVertex(vertexId);
+    while (executedVertexOpt.isPresent()) {
+      final var v = executedVertexOpt.unwrap();
+      path.addFirst(v);
+      if (v.getVertexHash().equals(v.vertex().getParentVertexId())
+          || v.getVertexHash().equals(this.rootVertex.hash())) {
+        break;
```

Comment:
Shouldn't it be a `return path` here, and we should throw an Exception (or return None) if we execute the path via an ExecutedVertex being `Some`?

(other than if `vertexId = root`, obvs)

Reply:
hmh not sure I get it, `return path` is equivalent to `break` here? why would we throw if executed vertices are some?

Reply:
Well if we don't find a vertex on the path to the root we simply return an incomplete path at the moment.

This feels wrong / buggy?

Reply:
Can we also possibly add a JavaDoc comment to explain that the path doesn't contain the root vertex? :)

Reply:
ah, but I think that's needed for a scenario when ledger is ahead of vertex store